be/src/vec/olap/vgeneric_iterators.cpp

```diff
@@ -382,21 +399,36 @@ Status VMergeIterator::next_batch(vectorized::Block* block) {
         _block_row_locations.resize(block_row_max);
     }
     size_t row_idx = 0;
+    VMergeIteratorContext* pre_ctx = nullptr;
     while (block->rows() < block_row_max) {
         if (_merge_heap.empty()) break;
 
         auto ctx = _merge_heap.top();
```

Comment:
We should make this comparison inside one block as much as possible instead of always using a heap to compare rows between different blocks.